include/alp/omp/matrix.hpp

```diff
@@ -52,41 +55,45 @@ namespace alp {
 			> * = nullptr
 		>
 		typename internal::new_container_type_from<
-			typename SourceMatrix::template view_type< view::cross_backend >::type
-		>::template change_backend< reference >::type
+			typename SourceMatrix::template view_type< view::gather >::type
+		>::template change_backend< config::default_sequential_backend >::type
 		get_view( SourceMatrix &source, const size_t tr, const size_t tc, const size_t rt, const size_t br, const size_t bc ) {
 
 			(void) rt;
 
-			using value_type = typename SourceMatrix::value_type;
-
 			// get the container
 			const auto &distribution = getAmf( source ).getDistribution();
```

Comment:
How do we make sure that the amf is distributed? Perhaps we should statically check based say on its backend?